src/server/bg_services/namespace_monitor.js

```diff
@@ -214,12 +220,21 @@ class NamespaceMonitor {
             await nb_native().fs.readdir(fs_context, nsr.nsfs_config.fs_root_path);
             const stat = await nb_native().fs.stat(fs_context, nsr.nsfs_config.fs_root_path);
             //In the event of deleting the nsr.nsfs_config.fs_root_path in the FS side, 
-            // The number of link will be 0, then we will throw ENOENT which translate to STORAGE_NOT_EXIST
+            // The number of link will be 0, then we will throw an error which translate to STORAGE_NOT_EXIST
             if (stat.nlink === 0) {
+                dbg.log1('test_nsfs_resource: the number of links is 0');
                 throw Object.assign(new Error('FS root path has no links'), { code: 'ENOENT' });
             }
         } catch (err) {
-            dbg.log1('test_nsfs_resource: got error:', err, nsr.nsfs_config);
+            dbg.error('test_nsfs_resource: got error:', err, nsr.nsfs_config);
+            // we change the code to control the mapping in pool server when calc the namespace mode
+            if (err.code === 'ENOENT') {
+                // it can happen if (1) stat/readdir threw ENOENT or (2) the number of links is 0 
+                throw Object.assign(err, { code: S3Error.NoSuchBucket.code });
+            }
+            if (err.code === `EPERM` || `EACCES`) {
```

Comment:
err.code === `EPERM` || `EACCES` always returns true because of the 'EACCES' - 
should be (err.code === 'EPERM' || err.code === 'EACCES')

Reply:
My mistake, thank you.